lib/spack/spack/test/sbang.py

```diff
@@ -191,18 +191,11 @@ def test_shebang_handles_non_writable_files(script_dir, sbang_line):
 
 def check_sbang_installation():
     sbang_path = sbang.sbang_install_path()
-    sbang_bin_dir = os.path.dirname(sbang_path)
     assert sbang_path.startswith(spack.store.store.unpadded_root)
 
     assert os.path.exists(sbang_path)
     assert fs.is_exe(sbang_path)
 
-    status = os.stat(sbang_path)
-    assert (status.st_mode & 0o777) == 0o755
-
-    status = os.stat(sbang_bin_dir)
-    assert (status.st_mode & 0o777) == 0o755
-
```

Comment:
Assuming `sbang_bin_dir` permissions update is retained, this check should also be restored.

Reply:
Here we could instead check the config to see if 0o755 or 0o775 should be set (see if a group is specified), then check if it as actually set correctly.

Reply:
Whatever makes sense.

Reply:
My point was that it would be good to make sure the directory has suitable permissions.

Reply:
@tldahlgren I added this test in bb9299d